keps/sig-node/1797-configure-fqdn-as-hostname-for-pods/README.md

```diff
@@ -708,18 +714,23 @@ _This section must be completed when targeting beta graduation to a release._
   It is not affected.
 
 * **What are other known failure modes?**
-  TODO
 
   For each of them fill in the following information by copying the below template:
-  - [Failure mode brief description]
+  - Pod FQDN is longer than 64 bytes
     - Detection: How can it be detected via metrics? Stated another way:
-      how can an operator troubleshoot without loogging into a master or worker node?
+      how can an operator troubleshoot without logging into a master or worker node?
+      Pods configured to obtain FQDN that make use of this feature will remain in Pending status generating error events regarding failure to create PodSandbox due to too long FQDN. We could use the metric `run_podsandbox_errors_total` to identify abnormal number of failures creating PodSandbox.
+
     - Mitigations: What can be done to stop the bleeding, especially for already
       running user workloads?
+      Pods should either set the PodSpec field `setHostnameAsFQDN`.
+
     - Diagnostics: What are the useful log messages and their required logging
       levels that could help debugging the issue?
-      Not required until feature graduated to Beta.
+      This issue will be logged in Error level log messages. The error will something like`GeneratePodSandboxConfig for pod foo failed: Failed to construct FQDN from pod hostname and cluster domain, FQDN <long-fqdn> is too long (64 characters is the max, 70 characters requested)`      
+
     - Testing: Are there any tests for failure mode? If not describe why.
+      Unittests cover this failure scenario. The e2e tests also check for this error, but only as protection for the CI. A test that purposely hit this issue will be flaky as the pod will remain in Pending status logging errors until the test timesout.
```

Comment:
I don't understand this - why the test should be flaky?

The test could be:
- create a pod (that would exceed this limit)
- watch for events and ensure the even for fdqn is delivered
- once event is delivered, delete the pod

We have similar kind of tests and I don't think they are really flaky.

Reply:
Cool. Once done, please update this part here.

Reply:
Please update - the "flaky" part doesn't make sense to me (as I wrote above). You can describe that test in testing section or sth like that.

Reply:
Yes, I will update the section saying e2e also covers failure mode so I don't forget.
However, I am still working to get the e2e test right, I am working based on an AppArmor e2e test I found, i think it does what you suggested. https://github.com/kubernetes/kubernetes/pull/95187